### PR TITLE
    Check for allocator bucket overflow

### PR DESCRIPTION
**Pull request by wlepera**
Ensure requested memory size can be accommodated by the current number
    of memory allocator buckets

Signed-off-by: William P. LePera <lepera@us.ibm.com>

### CODE REVIEW

File: opal/mca/allocator/bucket/allocator_bucket_alloc.c [hunk 1]
```diff
@@ -89,6 +94,13 @@ void *mca_allocator_bucket_alloc(mca_allocator_base_module_t *mem, size_t size)
         bucket_size <<= 1;
     }
 
+    if( bucket_num > max_bucket_idx ) {
+        size_t sz_bucket = MCA_ALLOCATOR_BUCKET_1_SIZE;
+        fprintf (stderr, "ERROR: Requested buffer size %zu exceeds limit of %zu\n", size, sz_bucket << max_bucket_idx);
+        fprintf (stderr, "Consider setting \"allocator_bucket_num_buckets\" to %d\n", bucket_num + 1);
```
Comment (jsquyres) [inline opal/mca/allocator/bucket/allocator_bucket_alloc.c:100]: These `fprintf()`s should probably be `opal_show_help()`s.

Comment (wlepera) [reply opal/mca/allocator/bucket/allocator_bucket_alloc.c]: done - fprintfs replaced with opal_show_help

**Comment by jsquyres** [comment]
bot:ompi:retest

**Comment by awlauria** [comment]
@wlepera can you squish this down into a single commit?

**Comment by ibm-ompi** [comment]
The IBM CI (PGI) build failed! Please review the log, linked below.

 Gist: https://gist.github.com/ibm-ompi/88b480fb98dfd2bdac939f023e1f8fe6

**Review by awlauria — APPROVED** [review]

**Comment by awlauria** [comment]
LGTM

**Pull request merged**